### DISCUSSION TITLE
unbind events

### DISCUSSION THREAD

**Discussion by CrystalWindSnake** [zauberzeug/nicegui#1516] [Ideas / Feature Requests]
### Description

How can I achieve the following effect?
```python
label = ui.label('click me!').on('click',lambda:print('click label'))

def onclick():
    pass
    # how to unbind label click event

ui.button('unbind label click event',on_click=onclick)
```

I noticed that in the base class element, every time an event is bound using the on function, it creates an EventListener and stores it in a dictionary.but it seems that there is no mechanism to unbind the already bound event.

**Comment by falkoschindler** [comment]
That's right, there is currently no way to unbind an event handler.
See https://github.com/zauberzeug/nicegui/discussions/345 for a related discussion.

As a workaround you can remove and recreate the element.

But we usually simply keep the event handler and handle it accordingly:

```py
state = {'can_click': True}

def handle_click():
    if state['can_click']:
        ui.notify('click')

ui.label('click me!').on('click', handle_click)

ui.button('deactivate click', on_click=lambda: state.update(can_click=False))
```

**Reply by CrystalWindSnake** [reply]
I currently have some animation needs and will bind one-time events to many elements. It would be great if nicegui can provide a good API for canceling events. I look forward to future support for canceling events in nicegui.

**Comment by rodja** [comment]
I'll convert this to a feature request. Anyone interested in creating a pull request?
